api/base/parsers.py

```diff
@@ -15,6 +17,29 @@ class JSONAPIParser(JSONParser):
     media_type = 'application/vnd.api+json'
     renderer_class = JSONAPIRenderer
 
+    def flatten_relationships(self, relationships):
+        """
+        Flattens relationships dictionary which has information needed to create related resource objects.
+
+        Validates that formatting of relationships dictionary is correct.
+        """
+        if not isinstance(relationships, dict):
+                raise ParseError()
+        related_resource = relationships.keys()[0]
```

Comment:
Hmm, I think for the /contributors and /node_links/ URL's, there would only be one key.  The only type of relationship you can be creating under the /v2/nodes/{node_id}/contributors/ would be the relationship between the node and the contributor, and /v2/nodes/{node_id}/node_links? would be the node and target node. These two URL's describe collections of relationships.

However, at some point we might implement creating relationships where the URL does not describe a relationships collection but a resource collection. There you might want to have multiple items under relationships.  Say I'm posting to /v2/nodes/ and simultaneously want to add a node link, and a contributor.  So I'm creating a node, that has a relationship between a contributor, and another node, in one sweep. We're not doing this, but you could.